src/pip/_internal/cli/cmdoptions.py

```diff
@@ -25,6 +26,25 @@
     from typing import Any  # noqa: F401
 
 
+class OptionValues(optparse.Values):
```

Comment:
Would a cleaner approach be to have a custom action callback function instead? This case can definitely be easily satisfied with an Option `callback` instead and a function will be "lighter weight" than this subclass.

https://docs.python.org/3/library/optparse.html#optparse-option-callbacks

Reply:
Good thought! The special handling we need here takes place in the `_update_defaults()` method: https://github.com/pypa/pip/blob/3b8c076c1fed4ec52d34aa6f8bc4739a4e778b47/src/pip/_internal/cli/parser.py#L176
rather than in the usual option-parsing itself, but it looks like we can use "callback" there as well (there is already code for that case). I'll investigate!

PS - I'm going to delete your 6 or so duplicate comments that must have been caused by the outage..

Reply:
> PS - I'm going to delete your 6 or so duplicate comments that must have been caused by the outage..

I got confused by what was happening. :P